web/app/datasets/external/sigmax.py

```diff
@@ -23,20 +87,64 @@ def _format_date(dt):
     return dt.strftime('%Y%m%d')
 
 
+class ServiceNotConfigured(Exception):
+    pass
+
+
 # -- functions that generate or send messages --
 
 def _generate_stuf_message(signal):
     """
     Generate the XML needed for Sigmax.
     """
-    pass
+    return TEMPLATE.format(**{
+        'PRIMARY_KEY': escape(signal['signal_id']),
+        'OMSCHRIJVING': escape('Dit is een test bericht'),
+        'TIJDSTIPBERICHT': escape(PLACEHOLDER_STRING),
+        'STARTDATUM': escape(PLACEHOLDER_STRING),
+        'REGISTRATIEDATUM': escape(PLACEHOLDER_STRING),
+        'EINDDATUMGEPLAND': escape(PLACEHOLDER_STRING),
+        'OPENBARERUIMTENAAM': escape(PLACEHOLDER_STRING),
+        'HUISNUMMER': escape(PLACEHOLDER_STRING),
+        'POSTCODE': escape(PLACEHOLDER_STRING),
+    })
 
 
 def _send_stuf_message(stuf_msg):
     """
     Send a STUF message to the server that is configured.
     """
-    pass
+    # Grab credentials from environment (assumption, these are set for
+    # either testing or production --- not configurable at run time).
+    SIGMAX_AUTH_TOKEN = os.getenv('SIGMAX_AUTH_TOKEN', None)
+    SIGMAX_SERVER = os.getenv('SIGMAX_SERVER', None)
+    logger.debug('SIGMAX_SERVER: {}'.format(SIGMAX_SERVER))
+    logger.debug('SIGMAX_AUTH_TOKEN: {}'.format(SIGMAX_AUTH_TOKEN))
+
+    if not SIGMAX_AUTH_TOKEN or not SIGMAX_SERVER:
+        msg = 'SIGMAX_AUTH_TOKEN or SIGMAX_SERVER not configured.'
```

Comment:
Should this not be checked at startup of the application? Better to fail during start then wait for failure when sending.

Reply:
The idea here was that the service can run also when not all credentials for all services are present (given that this service may talk to a number of external services). It may be a good idea to also add a start-up check (but that could possibly just be a bash script with `set -u` that goes over the env variables), but I would like to retain this local check as well so that we get a clear error message for this specific error condition.